src/vs/platform/extensionManagement/common/extensionTipsService.ts

```diff
@@ -61,15 +61,9 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 				continue;
 			}
 			try {
-				let content: string | undefined;
-				const getContent = async () => {
-					if (content === undefined) {
-						content = (await this.fileService.readFile(joinPath(folder, configPath))).value.toString();
-					}
-					return content;
-				};
+				const content = (await this.fileService.readFile(joinPath(folder, configPath))).value.toString();
```

Comment:
this file read will no longer be short circuited if `value.contentPattern` is falsy

Reply:
Yes, this is expected. When `value.contentPattern` is falsy, it is enough for file to exist. When file does not exist, `readFile` throws error.